server/src/main/java/io/crate/analyze/CopyFromParserProperties.java

```diff
@@ -39,29 +41,47 @@
     public static final CopyFromParserProperties DEFAULT = CopyFromParserProperties.of(Settings.EMPTY);
 
     private final boolean emptyStringAsNull;
+    private final char columnSeparator;
 
     public static CopyFromParserProperties of(Settings settings) {
+        var csvColumnSeparatorAsString = settings.get(CSV_COLUMN_SEPARATOR.getKey());
+        char csvColumnSeparator;
+        if (csvColumnSeparatorAsString == null) {
+            csvColumnSeparator = CsvSchema.DEFAULT_COLUMN_SEPARATOR;
```

Comment:
Could we set the default value in the `CSV_COLUMN_SEPARATOR` definition?

It might also be possible to change the type of the setting to  `Setting<Char>`, then the whole logic could be encapsulated within the setting itself and here we'd use `CSV_COLUMN_SEPARATOR.get(settings)` to retrieve the final value.

Reply:
But the default value is always going to have the `String` type, so I am not sure whether we can do it without changing the `Setting.class`.
 
```
    private Setting(Key key, @Nullable Setting<T> fallbackSetting, Function<Settings, String> defaultValue, Function<String, T> parser,
                    Validator<T> validator, Property... properties) {
```

Reply:
Something like this should work, no?


```java
public static final Setting<Character> CSV_COLUMN_SEPARATOR = new Setting<Character>(
    "delimiter",
    ",",
    value -> {
        if (value.length() != 1) {
            throw new IllegalArgumentException(
                "Invalid CSV fields delimiter: " + value + ". The delimiter must be a single character.");
        }
        return value.charAt(0);
    },
    Setting.Property.Dynamic
);
```

Reply:
hmm indeed.. was confused about default string value.